lib/queuePopulator/LogReader.js

```diff
@@ -522,6 +484,94 @@ class LogReader {
         return undefined;
     }
 
+    _processLogEntry(batchState, record, entry, cb) {
+        function _executeFilter(ext, entry, cb) {
+            if (typeof ext.filterAsync === 'function') {
+                ext.filterAsync(entry, cb);
+            } else {
+                ext.filter(entry);
+                process.nextTick(cb);
```

Comment:
can we make this more seamless and support async methods, by just checking the return?

e.g. maybe something like this:
```
res = ext.filter(entry)
if (res instanceof Promise) {
   res.then(cb)
} else {
   process.nextTick(cb);
}
```

And then maybe we do not even need to switch to callbacks in the extensions (or support filterAsync)....